gc/base/standard/ParallelGlobalGC.cpp

```diff
@@ -1120,6 +1151,12 @@ MM_ParallelGlobalGC::internalPostCollect(MM_EnvironmentBase *env, MM_MemorySubSp
 #endif /* defined (OMR_GC_LARGE_OBJECT_AREA) */
 
 
+	OMRPORT_ACCESS_FROM_ENVIRONMENT(env);
+	omrtty_printf("Current gc code is: %d\n", env->_cycleState->_gcCode.getCode());
+	if (env->_cycleState->_gcCode.isExplicitGC()){
```

Comment:
will have to call it earlier, just before a set of reportGC* calls  
the point is that heap-fixup in VGC is reported by CycleEnd. event, so it has to be before it to be properly reported.

Reply:
you can see in VGC code in MM_VerboseHandlerOutput::handleCycleEnd how we consume fixHeapForWalkReason there

also if you search for J9HOOK_MM_PRIVATE_GC_POST_CYCLE_END you can see that fixup reason and duration are parts of that hook event and how we pass that information when we trigger that event

Reply:
I see, so incorrect order will cause the incorrect vgc output